main/core/src/define/Ctx.scala

```diff
@@ -11,7 +11,7 @@ sealed trait Segment{
 }
 object Segment{
   case class Label(value: String) extends Segment{
-    assert(!value.contains('.'))
+    assert(!value.contains('.'), s"$value contained a .")
```

Comment:
```suggestion
    require(!value.contains('.'), s"A label must not contain a '.' (actual value: '$value'")
```

An assert is probably not the best choice here?